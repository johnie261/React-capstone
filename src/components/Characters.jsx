import React, { useEffect } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { fetchCharacters } from '../redux/character/characterSlice';
import Character from './Character';
import styled from 'styled-components';
import { Link } from 'react-router-dom';
import Details from '../pages/details';

const Characters = ({characters}) => {
    const { isLoading } = useSelector((store) => store.character);
    console.log(isLoading)
    const dispatch = useDispatch();

    useEffect(()=>{
        dispatch(fetchCharacters());
    }, [])

    if(isLoading) {
      return (
        <div className="sect">
          <h2>Loading...</h2>
        </div>
      )
    }

  return (
    <Wrapper>   
    <p className="title-character">STATS BY CHARACTER</p>
    <ul className="characters">
     
        {characters.map((character)=>(
          <Link to={`details/${character.id}`} element={<Details />} className="link">
            <Character
            key={character.id}
            id={character.id}
            image={character.image}
            name={character.name}
            />
          </Link>
        )
        )}
    </ul>
    </Wrapper>
  )
}

const Wrapper = styled.section`
background-color: #ec4c8b;
background-color: #db3578;
.title-character {
  color: #fff;
  font-size: 1.7rem;
  padding: 2rem;
  margin-top: -2rem;
}
.characters {
  margin-top: -3rem;
  list-style: none;
  padding: 3rem;
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(20%, 1fr));
  grid-template-columns: 1fr 1fr 1fr 1fr;
  gap: 1rem;
  .link {
    text-decoration: none;
  }
}

@media only screen and (max-width: 768px) {
  .title-character {
    font-size: 1.3rem;
  }
  .characters {
    grid-template-columns: 1fr 1fr 1fr;
  }
}

@media only screen and (max-width: 500px) {
  .title-character {
    font-size: 1.3rem;
  }
  .characters {
    grid-template-columns: 1fr 1fr;
  }
}

`

export default Characters