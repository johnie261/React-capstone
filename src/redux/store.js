import { configureStore } from '@reduxjs/toolkit';
import characterReducer from './character/characterSlice';

const store = configureStore({
    reducer: {
        character: characterReducer,
    }
})

export default store;