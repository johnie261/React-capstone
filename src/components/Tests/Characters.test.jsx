import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Characters from '../Characters';
import store from '../../redux/store';

describe('Characters component', () => {
  it('matches the snapshot', () => {
    const characters = [
      {
        id: 1,
        name: 'Harry Potter',
        image: 'harry.jpeg',
      },
      {
        id: 2,
        name: 'Hermione Granger',
        image: 'hermione.jpeg',
      },
    ];
    const tree = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Characters characters={characters} />
        </BrowserRouter>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
