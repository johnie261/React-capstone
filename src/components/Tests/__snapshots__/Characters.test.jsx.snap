// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Characters component matches the snapshot 1`] = `
<section
  className="sc-eDDNvR hspNbr"
>
  <p
    className="title-character"
  >
    STATS BY CHARACTER
  </p>
  <ul
    className="characters"
  >
    <a
      className="link"
      element={<Details />}
      href="/details/1"
      onClick={[Function]}
    >
      <div
        className="sc-beqWaB jtKaYx"
      >
        <li
          className="character"
        >
          <img
            alt="Harry Potter"
            className="image"
            src="harry.jpeg"
          />
          <div
            className="txt"
          >
            <p>
              Harry Potter
            </p>
          </div>
        </li>
      </div>
    </a>
    <a
      className="link"
      element={<Details />}
      href="/details/2"
      onClick={[Function]}
    >
      <div
        className="sc-beqWaB jtKaYx"
      >
        <li
          className="character"
        >
          <img
            alt="Hermione Granger"
            className="image"
            src="hermione.jpeg"
          />
          <div
            className="txt"
          >
            <p>
              Hermione Granger
            </p>
          </div>
        </li>
      </div>
    </a>
  </ul>
</section>
`;
